webui/react/src/components/JupyterLabModal.tsx

```diff
@@ -349,12 +349,23 @@ const JupyterLabForm: React.FC<{
 }> = ({ form, formId, currentWorkspace, defaults, lockedWorkspace, setWorkspace, workspaces }) => {
   const [templates, setTemplates] = useState<Template[]>([]);
 
-  const resourcePools = Loadable.getOrElse([], useObservable(clusterStore.resourcePools));
+  const selectedWorkspace = Form.useWatch('workspaceId', form);
+
+  const resourcePools = useObservable(clusterStore.resourcePools);
+  const boundResourcePoolsMap = useObservable(workspaceStore.boundResourcePoolsMap());
+
+  const boundResourcePools: ResourcePool[] = useMemo(() => {
+    if (!Loadable.isLoaded(resourcePools) || !boundResourcePoolsMap || !selectedWorkspace)
```

Comment:
what does `!boundResourcePoolsMap` mean here? i dont think `boundResourcePoolsMap` can be `undefined`. and empty map still return true.

nit: `selectedWorkspace === undefined`